### DISCUSSION TITLE
Templating in icon and icon_color for cover card

### DISCUSSION THREAD

**Discussion by johannes-z** [piitaya/lovelace-mushroom#449] [Ideas]
It would be really helpful to be able to set dynamic icons and icon_color for the cover card. I have a switch that locks my covers, preventing them to be moved up/down (as to not lock yourself out of the house). It would be cool if I could change the icon/icon_color of the cover card whenever the switch comes on, saving some space on the dashboard. It would basically combine the following into one control:

![image](https://user-images.githubusercontent.com/9069888/167272802-4b6bdafb-0a2b-4dcb-802c-0badf855fc5b.png)

When *Shutter Lock* is `on`, I'd want to change the icon to `mdi:lock` and set the icon_color to e.g. red.

**Comment by piitaya** [comment]
It's not planned to add template to other card that template card to keep mushroom simple. 
As a fix, you can use a condition card to display a template card instead the cover card. The button doesn't make sense if your entity is locked.

**Reply by thesebastianf** [reply]
`As a fix, you can use a condition card to display a template card instead the cover card. The button doesn't make sense if your entity is locked.`

this sounds like what I want :)
only show another mushroom card type when a condition applies. (the light card has no condition)
not sure how this is done however. A condition card is a template card? How can I nest a light card within?
thanks

**Comment by Gitonthegrid** [comment]
I don't know what you can and can`t do with the cover card, But I use this for my template cards to change color and icons depending on a state:

**For icon change, put this in the icon section of the card** (change the sections your_entity_here / on or off for the state you want / and your_icon_here)
`{% if is_state('your_entity_here', 'on') %} mdi:your_icon_here
  {% elif is_state('your_entity_here', 'off') %} mdi:your_icon_here
  {% else %} mdi:your_icon_here
  {% endif %}`

**Put this in the icon color section** (change the sections your_entity_here / on or off for the state you want / and the blue and red for the colors you want)
`{% if is_state('your_entity_here', 'on') %} blue
  {% elif is_state('your_entity_here', 'off') %} red
  {% else %} purple
  {% endif %}`
